source/common/http/utility.cc

```diff
@@ -657,6 +657,16 @@ Utility::resolveMostSpecificPerFilterConfigGeneric(const std::string& filter_nam
   return maybe_filter_config;
 }
 
+std::vector<const Router::RouteSpecificFilterConfig*>
```

Comment:
perf nit: you might consider either an std::array of size 3 here, given that I think we know there can be a max of 3 entries? Or potentially for easier programming an absl::InlineVector of size 3 default storage?

Reply:
ooh, `absl::InlineVector` looks pretty cool. Although since we're holding pointers, callers should ideally be verifying non-`nullptr` anyways (as [the cors filter does](https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/cors/cors_filter.cc#L132)) so I think `std::array` would suffice

Reply:
still need to finish addressing this before final review